proxy/coco/src/peer/run_state.rs

```diff
@@ -608,10 +627,18 @@ impl RunState {
             },
             (_, RequestInput::Requested(urn, time, maybe_sender)) => {
                 let maybe_request = self.waiting_room.request(&urn, time);
+                let request = match maybe_request {
+                    Some(req) => req,
+                    None => self
```

Comment:
It'd be better to just return `Either` from `request` so that we don't have this `expect` business.

Reply:
I guess you mean `Option` here. #nohaskell And it wouldn't be better, as the caller then has to do another roundtrip to get the created request, rather argue that `waiting_room.request` ought to return a request in any case and not an `Option`.

Reply:
I think we're misunderstanding each other. The point I was trying to make is that request could return Either which is either the new request that is created or the existing request. That is, don't return Option here and return Either instead. This would mean we don't need to call get in the other branch. We would just call `into` to turn the new Created request into a SomeRequest.

Is that clearer or am I missing something?

Reply:
> Is that clearer or am I missing something?

Indeed, such clarification much types. Thanks that makes sense.

Reply:
Types as documentation 👌